### PR TITLE
Join group by day code

### LINKED ISSUE #442: Join small group via attendance code
At this moment, a user with no small group entering an attendance code for a small group results in an error. I think it would be logical, especially for meeting 1, if this added them to the small group the code was for

### PR DESCRIPTION
**Pull request by yongabyte**
fixed #442

### CODE REVIEW

File: server/api/attendance/attendance.controller.js [hunk 1]
```diff
@@ -365,20 +365,56 @@ exports.attend = function(req,res){
...
     	  .select('+dayCodes.code')
         .exec(function(err, smallgroup){
-          if (err) {return handleError(err)}
-          // if the user has no smallgroup, they cannont submit smallgroup  attendance
+          if (err) {return handleError(err)}          
+          
+          //if the user has no smallgroup,try comparing the code submission with the lastest dayCodes
           if (!smallgroup){
-              return res.status(400).json('No small group found or incorrect daycode!');
+              SmallGroup.findOne({"dayCodes.code":code,"classYear":classYear._id})
```
Comment (agundy) [inline server/api/attendance/attendance.controller.js]: could we break this out into a helper function? it is a lot of logic and code to be embedded in the controller

Comment (yongabyte) [inline server/api/attendance/attendance.controller.js]: you mean make a helper function for the database queries or wrap everything up after "if (!smallgroup)" and put it in a helper function ?

**Comment by agundy** [comment]
@kmcnellis what are your thoughts on making this more modular? I want to break this code up a tad because putting all the complex flow inside of a single controller function seems like a poor design choice.

File: server/api/attendance/attendance.controller.js [hunk 2]
```diff
@@ -365,20 +365,51 @@ exports.attend = function(req,res){
...
-      // Classyear attendance code and bonus code was incorrect, try small group
       else{
+      // Classyear attendance code and bonus code was incorrect, try small group        
         return SmallGroup.findOne({"students":user._id, "classYear":classYear._id})
     	  .select('+dayCodes.code')
         .exec(function(err, smallgroup){
-          if (err) {return handleError(err)}
-          // if the user has no smallgroup, they cannont submit smallgroup  attendance
+          if (err) {return handleError(err)}          
+
```
Comment (kmcnellis) [inline server/api/attendance/attendance.controller.js]: extra line

**Comment by kmcnellis** [comment]
I also think it would be best to split it into a extra function. That controller is complicated enough as is

File: server/api/attendance/attendance.spec.js [hunk 3]
```diff
@@ -1,9 +1,13 @@
 'use strict';
+import {seed} from '../../config/seed';
+import Smallgroup from '../smallgroup/smallgroup.model';
+import {getSessionFor} from '../../auth/util.integration';
 
 var should = require('should');
 var app = require('../../app');
 var request = require('supertest');
 
+
```
Comment (kmcnellis) [inline server/api/attendance/attendance.spec.js]: extra line

File: server/api/attendance/attendance.controller.js (hunk 1 above)
Comment (kmcnellis) [inline server/api/attendance/attendance.controller.js]: Put the entire ` if (!smallgroup){` block in a separate function

**Comment by kmcnellis** [comment]
Looks better. @agundy You want to finish the review?

**Comment by kmcnellis** [comment]
@agundy You want to finish the review?

**Comment by kmcnellis** [comment]
@yongabyte Can you fix the conflicts so we can merge?

**Comment by kmcnellis** [comment]
Thanks!

**Pull request merged**